frontend/dispatcher.lua

```diff
@@ -214,46 +385,93 @@ function Dispatcher:addSubMenu(menu, location, settings)
                         UIManager:show(items)
                     end,
                     hold_callback = function(touchmenu_instance)
-                        self[location][settings][k] = nil
+                        caller[location][settings][k] = nil
                         if touchmenu_instance then
                             touchmenu_instance:updateItems()
                         end
                     end,
+                    separator = settingsList[k].separator,
                 })
             elseif settingsList[k].category == "string" then
                 local sub_item_table = {}
                 for i=1,#settingsList[k].args do
                     table.insert(sub_item_table, {
                         text = tostring(settingsList[k].toggle[i]),
                         checked_func = function()
-                            return self[location][settings] ~= nil
-                            and self[location][settings][k] ~= nil
-                            and self[location][settings][k] == settingsList[k].args[i]
+                            return caller[location][settings] ~= nil
+                            and caller[location][settings][k] ~= nil
+                            and caller[location][settings][k] == settingsList[k].args[i]
```

Comment:
I also don't like that, would prefer more indentation for the continuation lines.